From 18a91887266936d2e7df273df112b27a86e657d1 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 9 Apr 2024 20:00:06 +0200
Subject: [PATCH 1259/1695] dts: st: fix timers and lptimers node in stm32mp131

Device tree alignment with Linux kernel, add subnode for timer nodes.
Timer and LP Timer nodes now complies with the yaml dt-bindings.
DTC warnings messages solved:
"... Warning (avoid_unnecessary_addr_size) ... unnecessary ...
#address-cells/#size-cells without ... child "reg" property"

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7e8cbcdbec6fb5a70396c2eb26ae429c88d94af6 dts: st: fix timers and lptimers node in stm32mp131]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Iaca12f9a679a51408b9884c8c621a20cd0eff629
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/407690
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Olivier MOYSAN <olivier.moysan@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi | 135 ++++++++++++++++++++++++++++++
 1 file changed, 135 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index 72f803a16..b57a865b4 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -712,6 +712,7 @@
 				compatible = "st,stm32-timers";
 				reg = <0x4c007000 0x400>;
 				interrupts = <GIC_SPI 104 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "global";
 				clocks = <&rcc TIM12_K>;
 				clock-names = "int";
 				access-controllers = <&etzpc STM32MP1_ETZPC_TIM12_ID>;
@@ -720,6 +721,18 @@
 					compatible = "st,stm32-timer-counter";
 					status = "disabled";
 				};
+
+				pwm {
+					compatible = "st,stm32-pwm";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer@11 {
+					compatible = "st,stm32h7-timer-trigger";
+					reg = <11>;
+					status = "disabled";
+				};
 			};
 
 			timers13: timer@4c008000 {
@@ -727,10 +740,29 @@
 				#size-cells = <0>;
 				compatible = "st,stm32-timers";
 				reg = <0x4c008000 0x400>;
+				interrupts = <GIC_SPI 111 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "global";
 				clocks = <&rcc TIM13_K>;
 				clock-names = "int";
 				access-controllers = <&etzpc STM32MP1_ETZPC_TIM13_ID>;
 				status = "disabled";
+
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32-pwm";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer@12 {
+					compatible = "st,stm32h7-timer-trigger";
+					reg = <12>;
+					status = "disabled";
+				};
 			};
 
 			timers14: timer@4c009000 {
@@ -738,10 +770,29 @@
 				#size-cells = <0>;
 				compatible = "st,stm32-timers";
 				reg = <0x4c009000 0x400>;
+				interrupts = <GIC_SPI 112 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "global";
 				clocks = <&rcc TIM14_K>;
 				clock-names = "int";
 				access-controllers = <&etzpc STM32MP1_ETZPC_TIM14_ID>;
 				status = "disabled";
+
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32-pwm";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer@13 {
+					compatible = "st,stm32h7-timer-trigger";
+					reg = <13>;
+					status = "disabled";
+				};
 			};
 
 			timers15: timer@4c00a000 {
@@ -750,6 +801,7 @@
 				compatible = "st,stm32-timers";
 				reg = <0x4c00a000 0x400>;
 				interrupts = <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "global";
 				clocks = <&rcc TIM15_K>;
 				clock-names = "int";
 				access-controllers = <&etzpc STM32MP1_ETZPC_TIM15_ID>;
@@ -759,6 +811,18 @@
 					compatible = "st,stm32-timer-counter";
 					status = "disabled";
 				};
+
+				pwm {
+					compatible = "st,stm32-pwm";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer@14 {
+					compatible = "st,stm32h7-timer-trigger";
+					reg = <14>;
+					status = "disabled";
+				};
 			};
 
 			timers16: timer@4c00b000 {
@@ -766,11 +830,29 @@
 				#size-cells = <0>;
 				compatible = "st,stm32-timers";
 				reg = <0x4c00b000 0x400>;
+				interrupts = <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "global";
 				clocks = <&rcc TIM16_K>;
 				clock-names = "int";
 				access-controllers = <&etzpc STM32MP1_ETZPC_TIM16_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32-pwm";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer@15 {
+					compatible = "st,stm32h7-timer-trigger";
+					reg = <15>;
+					status = "disabled";
+				};
 			};
 
 			timers17: timer@4c00c000 {
@@ -778,10 +860,29 @@
 				#size-cells = <0>;
 				compatible = "st,stm32-timers";
 				reg = <0x4c00c000 0x400>;
+				interrupts = <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "global";
 				clocks = <&rcc TIM17_K>;
 				clock-names = "int";
 				access-controllers = <&etzpc STM32MP1_ETZPC_TIM17_ID>;
 				status = "disabled";
+
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32-pwm";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer@16 {
+					compatible = "st,stm32h7-timer-trigger";
+					reg = <16>;
+					status = "disabled";
+				};
 			};
 
 			lptimer2: timer@50021000 {
@@ -794,6 +895,28 @@
 				clock-names = "mux";
 				access-controllers = <&etzpc STM32MP1_ETZPC_LPTIM2_ID>;
 				status = "disabled";
+
+				pwm {
+					compatible = "st,stm32-pwm-lp";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				trigger@1 {
+					compatible = "st,stm32-lptimer-trigger";
+					reg = <1>;
+					status = "disabled";
+				};
+
+				counter {
+					compatible = "st,stm32-lptimer-counter";
+					status = "disabled";
+				};
+
+				timer {
+					compatible = "st,stm32-lptimer-timer";
+					status = "disabled";
+				};
 			};
 
 			lptimer3: timer@50022000 {
@@ -807,6 +930,18 @@
 				access-controllers = <&etzpc STM32MP1_ETZPC_LPTIM3_ID>;
 				status = "disabled";
 
+				pwm {
+					compatible = "st,stm32-pwm-lp";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				trigger@2 {
+					compatible = "st,stm32-lptimer-trigger";
+					reg = <2>;
+					status = "disabled";
+				};
+
 				counter {
 					compatible = "st,stm32-lptimer-counter";
 					status = "disabled";
-- 
2.47.2

